### PR TITLE
Add tests for local driver

### LINKED ISSUE #6841: Already dead when killing
This is a unit test for the local driver, which should test that the killing of an already terminated job is handled properly.

### PR DESCRIPTION
**Pull request by jonathan-eq**
Added local driver tests for testing that:
* killing an already dead job does not raise an exception.
* killing ERT does not kill its child processes

**Issue**
Resolves #6841 
Debunks #6844
## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/integration_tests/scheduler/test_integration_local_driver.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    assert expected_path.exists()
+    child_process_id = expected_path.read_text(encoding="utf-8").strip()
+    assert child_process_id
+
+    assert process.returncode is None
+    process.kill()
+    process.wait()
+    assert process.returncode is not None
+
+    # Child process should no longer exist
```
Comment (berland) [inline tests/integration_tests/scheduler/test_integration_local_driver.py:63]: This comment is wrong..?

Comment (jonathan-eq) [reply tests/integration_tests/scheduler/test_integration_local_driver.py:63]: Yes, it was outdated pretty quickly😬

File: tests/integration_tests/scheduler/test_integration_local_driver.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    ps_process = await asyncio.create_subprocess_exec("ps", "-p", child_process_id)
+    assert await ps_process.wait() == 0
```
[the remark this answers is not in the document]
Comment (jonathan-eq) [reply tests/integration_tests/scheduler/test_integration_local_driver.py:65]: This only checks if the child process is still running. If the process is dead/done, ps -p CHILD_PROCESS_ID returns non zero exitcode. I will see if killing the child process at the end of the test speeds it up, but I don't think so as it is already a separate process.

File: tests/integration_tests/scheduler/test_integration_local_driver.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+@pytest.mark.scheduler
+@pytest.mark.integration_test
+async def test_subprocesses_live_on_after_ert_dies(tmp_path, try_queue_and_scheduler):
+    # Have ERT run a forward model that writes in PID to a file, then sleeps
+    # Forcefully terminate ERT and assert that the child process is not terminated
+    create_ert_config(tmp_path)
+
+    process = subprocess.Popen(["ert", "test_run", "ert_config.ert"], cwd=tmp_path)
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_integration_local_driver.py:47]: ert_process

File: tests/integration_tests/scheduler/test_integration_local_driver.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    count, max_retry = 0, 50
+    expected_path = Path(tmp_path, "test_out/realization-0/iter-0/forward_model_pid")
+    while not expected_path.exists() and count < max_retry:
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_integration_local_driver.py:50]: check_path_retries

File: tests/unit_tests/scheduler/test_local_driver.py [hunk 5]
```diff
@@ -72,3 +72,16 @@ async def test_kill_when_job_completed(tmp_path, cmd, event):
     await asyncio.sleep(0.5)
     await driver.kill(42)
     assert await driver.event_queue.get() == (42, event)
+
+
+async def test_kill_already_killed_job(tmp_path):
```
[the remark this answers is not in the document]
Comment (xjules) [reply tests/unit_tests/scheduler/test_local_driver.py:77]: Maybe change the test name above too.

**Review by xjules — APPROVED** [review]
Nice job @jonathan-eq ! :rocket:

**Pull request merged**